app/controllers/smart_proxies_controller.rb

```diff
@@ -45,6 +45,13 @@ def tftp_server
     requested_data(:tftp_server)
   end
 
+  def puppet_environments
+    envs = @smart_proxy.public_send(:environment_stats)
```

Comment:
Actually this was just silly to use public_send here. I only need the data for the partial, `requested_data` gets data from the proxy and sends a json.